### PR TITLE
drivers: sensor: hp206c: Update driver to use DEVICE_DT_GET

### PR DESCRIPTION
**Pull request by galak**
Move driver to use DEVICE_DT_GET for handling access to bus parent
device struct.

Also removed unused gpio.h header include.

Signed-off-by: Kumar Gala <galak@kernel.org>

### CODE REVIEW

File: drivers/sensor/hp206c/hp206c.h [hunk 1]
```diff
@@ -73,7 +73,7 @@
 /* end of default settings */
 
 struct hp206c_device_data {
-	const struct device *i2c;
+	const struct device *i2c_bus_dev;
```
Comment (MaureenHelm) [inline drivers/sensor/hp206c/hp206c.h:76]: Why did you rename this? The change adds some unnecessary noise to the diff

**Review by MaureenHelm — CHANGES_REQUESTED** [review]

File: drivers/sensor/hp206c/hp206c.h (hunk 1 above)
Comment (galak) [reply drivers/sensor/hp206c/hp206c.h]: removed the variable name change.

File: drivers/sensor/hp206c/hp206c.c [hunk 2]
```diff
@@ -288,9 +287,10 @@ static int hp206c_init(const struct device *dev)
 {
 	struct hp206c_device_data *hp206c = dev->data;
 
-	hp206c->i2c = device_get_binding(DT_INST_BUS_LABEL(0));
-	if (!hp206c->i2c) {
-		LOG_ERR("I2C master controller not found!");
+	hp206c->i2c = DEVICE_DT_GET(DT_INST_BUS(0));
+
+	if (!device_is_ready(hp206c->i2c)) {
```
Comment (gmarull) [inline drivers/sensor/hp206c/hp206c.c:292]: we should be moving i2c to driver's config instead (using i2c_dt_spec), so that it gets stored in ROM

Comment (galak) [reply drivers/sensor/hp206c/hp206c.c]: Done!

File: drivers/sensor/hp206c/hp206c.h [hunk 3]
```diff
@@ -73,7 +73,7 @@
 /* end of default settings */
 
 struct hp206c_device_data {
-	const struct device *i2c;
+	const struct i2c_dt_spec i2c;
```
Comment (gmarull) [inline drivers/sensor/hp206c/hp206c.h:76]: should now be part of device config, so that it is const and goes to ROM.

File: drivers/sensor/hp206c/hp206c.h [hunk 4]
```diff
@@ -72,8 +72,11 @@
 #endif
 /* end of default settings */
 
+struct hp206c_device_config {
+	const struct i2c_dt_spec i2c;
```
Comment (gmarull) [inline drivers/sensor/hp206c/hp206c.h:76]:
```suggestion
	struct i2c_dt_spec i2c;
```

no need to make this const, as this is a struct, not a pointer. configuration constness comes from the definition of the config struct itself.

File: drivers/sensor/hp206c/hp206c.c [hunk 5]
```diff
@@ -318,6 +315,10 @@ static int hp206c_init(const struct device *dev)
 
 static struct hp206c_device_data hp206c_data;
 
+static struct hp206c_device_config hp206c_config = {
```
Comment (gmarull) [inline drivers/sensor/hp206c/hp206c.c:318]:
```suggestion
static const struct hp206c_device_config hp206c_config = {
```

**Review by gmarull — CHANGES_REQUESTED** [review]

**Review by gmarull — APPROVED** [review]

**Review by MaureenHelm — APPROVED** [review]
